### PR TITLE
Feature/issue 250

### LINKED ISSUE #250: Get the max borrowable amount through API
We need to show the max borrowable amount per NFT, but we cannot get it through API yet. 

I tried to implement it on the frontend, but as per Kimura san's guide, we should not implement those logics on the frontend, so would like to request it from API.

### PR DESCRIPTION
**Pull request by mkXultra**
fix #250

### CODE REVIEW

File: proto/nftmarket/query.proto [hunk 1]
```diff
@@ -120,6 +128,35 @@ message QueryNftBidsRequest {
...
+    (gogoproto.moretags)  = "yaml:\"bidder\"",
+    (cosmos_proto.scalar) = "github.com/UnUniFi/chain/types.StringAccAddress"
+  ];
+}
+
+message QueryPaymentStatusResponse {
+  PaymentStatus paymentStatus = 1 [(gogoproto.nullable) = false];
+}
+
+import "google/protobuf/timestamp.proto";
```
Comment (jununifi) [inline proto/nftmarket/query.proto:144]: I think import should be on top not in the middle.

File: x/nftmarket/keeper/grpc_query.go [hunk 2]
```diff
@@ -223,3 +255,49 @@ func (k Keeper) Rewards(c context.Context, req *types.QueryRewardsRequest) (*typ
...
 	return &types.QueryRewardsResponse{}, nil
 }
+
+func (k Keeper) PaymentStatus(c context.Context, req *types.QueryPaymentStatusRequest) (*types.QueryPaymentStatusResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+	_ = ctx
```
Comment (jununifi) [inline x/nftmarket/keeper/grpc_query.go:265]: This could be removed

File: x/nftmarket/keeper/grpc_query.go [hunk 3]
```diff
@@ -223,3 +255,49 @@ func (k Keeper) Rewards(c context.Context, req *types.QueryRewardsRequest) (*typ
...
+		if v.Bidder == req.Bidder {
+			bidderBid = v
+		}
+	}
+	if (bidderBid == types.NftBid{}) {
+		return nil, status.Error(codes.InvalidArgument, "does not match bidder")
+	}
+
+	allPaid := listing.State >= types.ListingState_END_LISTING && bidderBid.Amount.Amount.Equal(bidderBid.PaidAmount)
+	fmt.Println(allPaid)
```
Comment (jununifi) [inline x/nftmarket/keeper/grpc_query.go:290]: Log could be removed

File: x/nftmarket/keeper/grpc_query.go [hunk 4]
```diff
@@ -223,3 +255,49 @@ func (k Keeper) Rewards(c context.Context, req *types.QueryRewardsRequest) (*typ
...
+	return &types.QueryPaymentStatusResponse{
+		PaymentStatus: types.PaymentStatus{
+			NftId:            listing.NftId,
+			State:            listing.State,
+			Bidder:           bidderBid.Bidder,
+			Amount:           bidderBid.Amount,
+			AutomaticPayment: bidderBid.AutomaticPayment,
+			PaidAmount:       bidderBid.PaidAmount,
+			BidTime:          bidderBid.BidTime,
+			AllPaid:          false,
```
Comment (jununifi) [inline x/nftmarket/keeper/grpc_query.go:300]: allPaid flag should be returned correctly

File: proto/nftmarket/query.proto (hunk 1 above)
Comment (mkXultra) [reply proto/nftmarket/query.proto]: fixed

File: x/nftmarket/keeper/grpc_query.go (hunk 2 above)
Comment (mkXultra) [reply x/nftmarket/keeper/grpc_query.go]: fixed

File: x/nftmarket/keeper/grpc_query.go (hunk 3 above)
Comment (mkXultra) [reply x/nftmarket/keeper/grpc_query.go]: fixed

File: x/nftmarket/keeper/grpc_query.go (hunk 4 above)
Comment (mkXultra) [reply x/nftmarket/keeper/grpc_query.go]: fixed

**Pull request merged**